src/components/composer/content_editable_helper.ts

```diff
@@ -79,29 +90,61 @@ export class ContentEditableHelper {
   }
 
   /**
-   * Sets (or Replaces all) the text inside the root element in the form of distinctive
+   * Sets (or Replaces all) the text inside the root element in the form of distinctive paragraphs and
    * span for each element provided in `contents`.
    *
+   * The function will apply the diff between the current content and the new content to avoid the systematic
+   * destruction of DOM elements which interferes with IME[1]
+   *
+   * Each line of text will be encapsulated in a paragraph element.
    * Each span will have its own fontcolor and specific class if provided in the HtmlContent object.
+   *
+   * [1] https://developer.mozilla.org/en-US/docs/Glossary/Input_method_editor
    */
   setText(contents: HtmlContent[][]) {
-    this.el.innerHTML = "";
     if (contents.length === 0) {
+      this.removeAll();
       return;
     }
 
-    for (const line of contents) {
-      const p = document.createElement("p");
+    function compareContentToElement(content: HtmlContent, node?: HTMLElement | Text): boolean {
```

Comment:
Perosnal taste, but I feel like an inline function here don't bring anything and make the file harder to read. I'd put it in a private method.

Reply:
That was my first approach but then again I use really specific stuff like the tagName. I don't mind changing, I just struggled finding a proper name that would say "this contains SPAN specific logic"